VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/select-hearing-layout/select-hearing-layout.component.ts

```diff
@@ -25,9 +26,22 @@ export class SelectHearingLayoutComponent implements OnInit {
 
         (<any>window).GOVUKFrontend.initAll();
         headingElement.onclick = e => this.setAccordionText(e);
+        const sectionHeadingElement = document.getElementsByClassName('govuk-accordion__section-button').item(0) as HTMLButtonElement;
+        sectionHeadingElement.onclick = e => this.setAccordionText(e);
         this.accordionOpenAllElement = document.getElementsByClassName('govuk-accordion__open-all').item(0) as HTMLButtonElement;
         this.accordionOpenAllElement.onclick = e => this.setAccordionText(e);
         this.setAccordionText({} as MouseEvent);
+
+        this.translateService.onLangChange.subscribe(event => {
+            const updatedHeadingElement = document.getElementById('accordion-choose-layout-heading');
+            const currentHeaderText = updatedHeadingElement.innerText;
+            const updatedHeaderText = this.translateService.instant('select-hearing-layout.choose-hearing-layout');
+
+            updatedHeadingElement.innerHTML = updatedHeadingElement.innerHTML.replace(currentHeaderText, updatedHeaderText);
+            const currentTextValue = this.accordionOpenAllElement.innerText.split('\n')[0];
+            const translatedElement = this.translateService.instant(`select-hearing-layout.${this.currentButtonContentKey}`);
+            this.accordionOpenAllElement.innerHTML = this.accordionOpenAllElement.innerHTML.replace(currentTextValue, translatedElement);
```

Comment:
Does this not already get set by calling setAccordianText earlier in the function?